python/arcticdb/storage_fixtures/lmdb.py

```diff
@@ -0,0 +1,78 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file LICENSE.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import os
+import shutil
+from tempfile import mkdtemp
+from typing import Optional
+from itertools import chain
+from typing import Any
+
+from .api import *
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap
+from arcticdb.version_store.helper import add_lmdb_library_to_env
+
+# 128 MiB - needs to be reasonably small else Windows build runs out of disk
+_DEFAULT_CONFIG = {"map_size": 128 * (1 << 20)}
+
+
+class LmdbStorageFixture(StorageFixture):
+    def __init__(self, db_dir: Optional[str] = None):
+        super().__init__()
+        self.db_dir = str(db_dir) if db_dir else mkdtemp(suffix="LmdbStorageFixtureFactory")
+        self.arctic_uri = "lmdb://" + self.db_dir
+        self.libs_instances_from_arctic: List[NativeVersionStore] = []
+        self.arctic_instances: List[Arctic] = []
+
+    def create_arctic(self, **extras):
+        out = super().create_arctic(**extras)
+        out._accessed_libs = self.libs_instances_from_arctic
+        self.arctic_instances.append(out)
+        return out
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        for result in chain(self.libs_from_factory.values(), self.libs_instances_from_arctic):
+            #  pytest holds a member variable `cached_result` equal to `result` above which keeps the storage alive and
+            #  locked. See https://github.com/pytest-dev/pytest/issues/5642 . So we need to decref the C++ objects
+            #  keeping the LMDB env open before they will release the lock and allow Windows to delete the LMDB files.
+            result.version_store = None
+            result._library = None
+        for a in self.arctic_instances:
+            a._library_manager = None
+            pass
+
+        shutil.rmtree(self.db_dir)
+
+    def create_test_cfg(self, lib_name: str, *, lmdb_config: Dict[str, Any] = _DEFAULT_CONFIG) -> EnvironmentConfigsMap:
```

Comment:
`_DEFAULT_CONFIG` is mutable, which is dangerous for a kwarg

Reply:
It's never going to be mutated and there's no frozendict...

Reply:
The normal `if x is None: x = default` idiom would be fine